analyses/batch-effects/util/functions.R

```diff
@@ -0,0 +1,361 @@
+# function used to cut files
+shorten = function(df) {
+  return(df[1:100,])
+}
+
+# function used on kallisto data for summarizing transcripts at the gene level
+grouper = function(df){
+  df = df %>%
+    group_by(gene_id) %>%
+    summarise_at(colnames(df)[3:ncol(df)], mean, na.rm = TRUE)
+  return(df)
+}
+
+# transpose data through gather and spread functions, rename the index column
+trans = function(data, Kids_First_Biospecimen_ID, val, starting_col){
+  data = data %>%
+    gather(Kids_First_Biospecimen_ID, val, starting_col:(ncol(data))) %>%
+    spread("gene_id", val)
+  return(data)
+}
+
+# transpose the data and rename columns
+clean = function(data, gene_id){
+  data_t = trans(data, "id", "gene_expression", 2)
+  return(data_t)
+}
+
+# create directories and make sure paths will work on any computer
+handle_file_outputs = function(report_name){
+  # create output directory (if necessary)
+  dir.create(file.path(analysis_dir, "results"), showWarnings = FALSE)
+  
+  # create full base name of output file
+  report_name = file.path(analysis_dir, "results", report_name)
+  
+  return(report_name)
+}
+
+# make a PCA plot. Context name refers to the name that will be contactenated with the report name
+make_plot = function(tb, data_t, report_name, context_name){
+  data_t$batch = as_factor(data_t$batch)
+  shapes = c(2,3,11,12)
+  colors = c(
+    "#a6611a",
+    "#018571",
+    "#c2a5cf",
+    "#d01c8b",
+    "#e66101"
+  )
+  
+  p = ggplot(tb, aes(x = PC1, y=PC2, color=data_t$histology, shape=data_t$batch)) +
+    geom_point() +
+    scale_color_manual(values = colors) +
+    scale_shape_manual(values = shapes) +
+    labs(color = "Histology", shape = "batch") +
+    theme_bw() 
+  
+  ggsave(paste(report_name, context_name, sep = "_"))
+  
+  return(p)
+  
+  
+}
+
+# join batch information with gene expression and filter NA's
+tibble_combat_prep = function(df, gene_id, id_batch_histology){
+  
+  # clean the data (see clean comments)
+  data_t = clean(df, gene_id)
+  
+  # join batch information with gene expression data  
+  data_t = inner_join(id_batch_histology, data_t, by = "Kids_First_Biospecimen_ID")
+  
+  # filter out NAs
+  data_t = filter(data_t, !is.na(data_t$batch))
+  
+  return(data_t)
+}
+
+# for batchQC to work, the batches must be numeric values
+batch_prep = function(data_t){
+  # verify that batch is indeed a factor and numeric
+  batch = as_factor(data_t$batch)
+  batch = as.numeric(batch)
+  
+  # make so that batch values start at 1
+  levels(batch) <- 1:length(levels(batch))
```

Comment:
Could you use `as.integer()` instead and save some lines (untested)?

Reply:
I can't remember if I tested using as.integer(). I'll explore this option.